src/main/java/org/fundacionjala/virtualassistant/player/spotify/client/JsonNodeManagement.java

```diff
@@ -0,0 +1,96 @@
+    private final static String NO_TRACK_PLAYING = "No track is currently playing.";
+    private final ObjectMapper objectMapper;
+
+    public JsonNodeManagement (ObjectMapper objectMapper) {
+        this.objectMapper = objectMapper;
+    }
+
+    public String extractPlayerData(String playerData) throws TokenExtractionException {
+        StringBuilder result = new StringBuilder();
+
+        try {
+            JsonNode root = objectMapper.readTree(playerData);
+            String playbackType = root.path(FIELD_CURRENT_PLAYING_TYPE).asText();
+            result.append(RESULT_CURRENT_PLAYING_TYPE).append(playbackType).append(NEW_LINE);
+
+            if (TRACK.equals(playbackType)) {
+                JsonNode trackNode = root.path(FIELD_ITEM);
+                if (trackNode.isMissingNode()) {
+                    throw new MusicPlayerException(NO_TRACK_PLAYING);
+                }
+
+                String artistName = trackNode.path(FIELD_ARTISTS).get(0).path(FIELD_NAME).asText();
+                String albumName = trackNode.path(FIELD_ALBUM).path(FIELD_NAME).asText();
+                String trackName = trackNode.path(FIELD_NAME).asText();
+
+                result.append(RESULT_ARTISTS).append(artistName).append(NEW_LINE);
+                result.append(RESULT_ALBUM).append(albumName).append(NEW_LINE);
+                result.append(RESULT_NAME).append(trackName).append(NEW_LINE);
+            } else {
+                result.append(RESULT_MSG).append(playbackType).append(NEW_LINE);
+            }
+
+        } catch (Exception e) {
+            throw new TokenExtractionException(DATA_PLAYER_PROCESSING, e);
+        }
+
+        return result.toString();
+    }
+
+    public String extractTrackUriFromSearchResponse(String responseData) throws TokenExtractionException {
+        try {
+            JsonNode jsonNode = objectMapper.readTree(responseData);
+
+            if (jsonNode.has(FIELD_TRACKS) && jsonNode.get(FIELD_TRACKS).has(FIELD_ITEMS)) {
+                JsonNode items = jsonNode.get(FIELD_TRACKS).get(FIELD_ITEMS);
+                if (items.isArray() && !items.isEmpty()) {
+                    return items.get(0).get(URI).asText();
+                }
+            }
+        } catch (Exception e) {
+            throw new TokenExtractionException(DATA_PLAYER_PROCESSING, e);
+        }
+        return "";
+    }
+
+    public String extractCurrentTrackUri(String playerData) throws TokenExtractionException{
+        try {
+            JsonNode jsonNode = objectMapper.readTree(playerData);
+
+            if (jsonNode.has(FIELD_ITEM)) {
+                JsonNode item = jsonNode.get(FIELD_ITEM);
+                if (item.has(URI)) {
+                    return item.get(URI).asText();
+                }
+            }
+        } catch (IOException e) {
+            throw new TokenExtractionException(DATA_PLAYER_PROCESSING, e);
+        }
+        return null;
```

Comment:
never return a null value because this can cause null pointer exception, better to return an empty string